frontendUiTableImpl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/cell/BranchOrCommitCellRendererComponent.java

```diff
@@ -283,6 +284,15 @@ private static String getSyncToRemoteStatusBasedLabel(SyncToRemoteStatus status)
                 remoteName)));
   }
 
+  @UIEffect
+  private void setBranchToolTipText(IManagedBranchSnapshot branch) {
+    if (branch.isNonRoot()) {
+      setToolTipText(getSyncToParentStatusBasedToolTipText(branch.asNonRoot()));
+    } else if (branch.isRoot()) {
```

Comment:
The `if` is redundant; `!isNonRoot == isRoot`

Reply:
Nit: actually, you can rewrite the `if` in line 289 to check for `isRoot` instead of `isNonRoot` to avoid negation

Reply:
My bad. Was trying to use Vavr Match here but it didn't work as it is eager, so i returned to old if, but didn't refactor it properly.